packages/api/lib/cwSfExecutionEventUtils.js

```diff
@@ -23,57 +25,89 @@ const executionStatusToWorkflowStatus = (executionStatus) => {
+  return failures.pop();
+};
+
 /**
  * Get message to use for publishing failed execution notifications.
  *
- * Try to get the input to the first failed step in the execution so we can
+ * Try to get the input to the last failed step in the execution so we can
  * update the status of any granules/PDRs that don't exist in the initial execution
- * input. Falls back to overall execution input.
+ * input.
  *
- * @param {Object} inputMessage - Workflow execution input message
- * @returns {Object} - Execution step message or execution input message
+ * Falls back to overall execution input.
+ *
+ * @param {Object} inputCumulusMessage - Workflow execution input message
+ * @returns {Object} - CumulusMessage Execution step message or execution input message
  */
-const getFailedExecutionMessage = async (inputMessage) => {
+const getFailedExecutionMessage = async (inputCumulusMessage) => {
+  const amendedCumulusMessage = { ...inputCumulusMessage };
+
   try {
-    const executionArn = getMessageExecutionArn(inputMessage);
+    const executionArn = getMessageExecutionArn(amendedCumulusMessage);
     const { events } = await StepFunctions.getExecutionHistory({ executionArn });
 
-    const stepFailedEvents = events.filter(
-      (event) =>
-        ['ActivityFailed', 'LambdaFunctionFailed'].includes(event.type)
-    );
-    if (stepFailedEvents.length === 0) {
-      log.info(`No failed step events found in execution ${executionArn}`);
-      return inputMessage;
+    const lastFailedEvent = lastFailedEventStep(events);
+    if (lastFailedEvent.length === 0) {
+      log.warn(`No failed step events found in execution ${executionArn}`);
+      return amendedCumulusMessage;
     }
-
-    const lastStepFailedEvent = stepFailedEvents[stepFailedEvents.length - 1];
-    const failedStepExitedEvent = getStepExitedEvent(events, lastStepFailedEvent);
+    const failedExecutionStepName = getFailedStepName(events, lastFailedEvent.id);
+    const failedStepExitedEvent = getStepExitedEvent(events, lastFailedEvent);
 
     if (!failedStepExitedEvent) {
-      log.info(
-        `Could not retrieve output from last failed step in execution ${executionArn}, falling back to execution input`,
-        'Error:', new Error(`Could not find TaskStateExited event after step ID ${lastStepFailedEvent.id} for execution ${executionArn}`)
-      );
-
-      const exception = lastStepFailedEvent.lambdaFunctionFailedEventDetails
-        || lastStepFailedEvent.activityFailedEventDetails;
-
-      // If input from the failed step cannot be retrieved, then fall back to execution
-      // input.
-      return {
-        ...inputMessage,
-        exception,
+      log.info(`Could not retrieve output from last failed step in execution ${executionArn}, falling back to execution input`);
+      log.error(new Error(`Could not find TaskStateExited event after step ID ${lastFailedEvent.id} for execution ${executionArn}`));
```

Comment:
Because it was this before: 
```
      log.info(
        `Could not retrieve output from last failed step in execution ${executionArn}, falling back to execution input`,
        'Error:', new Error(`Could not find TaskStateExited event after step ID ${lastStepFailedEvent.id} for execution ${executionArn}`)
      );

```
And that didn't make any sense.  I suppose the info only is fine.

Reply:
I just changed it to a log.info() with all the same information, taking out the new Error weirdness.